### PR TITLE
fix(DB/SAI): Bogblossom trap - knockback and despawn.

### PR DESCRIPTION
**Pull request by avarishd**
Shouldn't the trapped and non trapped object be pooled?

## Changes Proposed:
- Add knockback and emote SAI for Bogblossom Bunny.
- Add despawn SAI for Bogblossom (Goober, that spawns the actual trap).

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/5163

## SOURCE:

https://github.com/TrinityCore/TrinityCore/commit/6fc2c62883b18be58201608848dadf348d1d91fe
## Tests Performed:

- tested

## How to Test the Changes:

Be a druid
.q a 10961
.go gameobject 20375
See if it knocksback and despawn after.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by BeefSteaker** [comment]
Tested, Knockback and emote work as expected

File: data/sql/updates/pending_db_world/rev_1681824132109329700.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+-- Bogblossom Bunny SAI (Source: https://www.youtube.com/watch?v=EIFhKHtvOe0) 
+UPDATE `creature_template` SET `AIName` = "SmartAI" WHERE `entry` = 23104;
+DELETE FROM `smart_scripts` WHERE `entryorguid` = 23104 AND `source_type` = 0;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`event_param5`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_param4`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(23104,0,0,1,54,0,100,0,0,0,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Bogblossom Bunny - On Just Summoned - Say Line 0"),
+(23104,0,1,0,61,0,100,0,0,0,0,0,0,11,40532,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Bogblossom Bunny - On Link - Cast 'Bogblossom Knockback'");
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1681824132109329700.sql:6]:
```suggestion
(23104,0,0,1,54,0,100,0,0,0,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,'Bogblossom Bunny - On Just Summoned - Say Line 0'),
(23104,0,1,0,61,0,100,0,0,0,0,0,0,11,40532,0,0,0,0,0,1,0,0,0,0,0,0,0,0,'Bogblossom Bunny - On Link - Cast \'Bogblossom Knockback\'');
```

File: data/sql/updates/pending_db_world/rev_1681824132109329700.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
+-- Bogblossom Bunny SAI (Source: https://www.youtube.com/watch?v=EIFhKHtvOe0) 
+UPDATE `creature_template` SET `AIName` = "SmartAI" WHERE `entry` = 23104;
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1681824132109329700.sql:2]:
```suggestion
UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 23104;
```

**Review by Gultask — APPROVED** [review]

**Comment by Gultask** [comment]
> Shouldn't the trapped and non trapped object be pooled?

Yeah probably. Remember to use 0-chanced pools tho https://github.com/azerothcore/azerothcore-wotlk/issues/15288 🙈

**Pull request merged**